pkg/rcmanager/ipinstance.go

```diff
@@ -0,0 +1,209 @@
+			err = m.localClusterRamaClient.NetworkingV1().RemoteVteps().Delete(context.TODO(), vtepName, metav1.DeleteOptions{})
+			if k8serror.IsNotFound(err) {
+				return nil
+			}
+		}
+		return err
+	}
+	vtepIP := node.Annotations[constants.AnnotationNodeVtepIP]
+	vtepMac := node.Annotations[constants.AnnotationNodeVtepMac]
+	instances, err := m.IPLister.List(labels.SelectorFromSet(labels.Set{constants.LabelNode: nodeName}))
+	if err != nil {
+		return err
+	}
+	newVtep := false
+
+	remoteVtep, err := m.remoteVtepLister.Get(vtepName)
+	if err != nil {
+		if !k8serror.IsNotFound(err) {
+			return err
+		}
+		remoteVtep = utils.NewRemoteVtep(m.ClusterName, m.RemoteClusterUID, vtepIP, vtepMac, node.Name, nil)
+		newVtep = true
+	}
+	curTime := metav1.Now()
+	remoteVtep = remoteVtep.DeepCopy()
+
+	remoteVtep.Spec.VtepIP = vtepIP
+	remoteVtep.Spec.VtepMAC = vtepMac
+	desired := func() []string {
+		ipList := make([]string, 0)
+		for _, v := range instances {
+			if v.Status.Phase == networkingv1.IPPhaseReserved {
+				continue
+			}
+			ip, _, _ := net.ParseCIDR(v.Spec.Address.IP)
+			ipList = append(ipList, ip.String())
+		}
+		return ipList
+	}()
+	actual := remoteVtep.Spec.EndpointIPList
+	desiredSet := gset.NewFrom(desired)
+	actualSet := gset.NewFrom(actual)
+	remove := actualSet.Diff(desiredSet)
+	add := desiredSet.Diff(actualSet)
+
+	remoteVtep.Spec.EndpointIPList = func() []string {
+		ipList := make([]string, 0, desiredSet.Size())
+		for _, v := range desiredSet.Slice() {
+			ipList = append(ipList, fmt.Sprint(v))
+		}
+		return ipList
+	}()
+	vtepChanged := vtepIP != "" && vtepMac != "" && (vtepIP != remoteVtep.Spec.VtepIP || vtepMac != remoteVtep.Spec.VtepMAC)
+	ipListChanged := remove.Size() == 0 || add.Size() == 0
+	if !newVtep && !ipListChanged && !vtepChanged {
+		return nil
+	}
+
+	if newVtep {
+		remoteVtep, err = m.localClusterRamaClient.NetworkingV1().RemoteVteps().Create(context.TODO(), remoteVtep, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	} else {
+		remoteVtep, err = m.localClusterRamaClient.NetworkingV1().RemoteVteps().Update(context.TODO(), remoteVtep, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	remoteVtep.Status.LastModifyTime = curTime
+	_, err = m.localClusterRamaClient.NetworkingV1().RemoteVteps().UpdateStatus(context.TODO(), remoteVtep, metav1.UpdateOptions{})
+	return err
+}
+
+func (m *Manager) nodeToIPInstance(nodeName string) ([]*networkingv1.IPInstance, error) {
+	podIP, err := m.IPIndexer.ByIndex(ByNodeNameIndexer, nodeName)
+	if err != nil {
+		klog.Warningf("[nodeToPodIP] can't use ipinstance indexer. indexername=%v, nodename=%v, err=%v", ByNodeNameIndexer, nodeName, err)
+		return nil, err
+	}
+	ans := make([]*networkingv1.IPInstance, 0)
+	for _, v := range podIP {
+		if instance, ok := v.(*networkingv1.IPInstance); ok {
+			ans = append(ans, instance)
+		}
+	}
+	return ans, nil
+}
+
+func (m *Manager) nodeToIPList(nodeName string) ([]string, error) {
```

Comment:
Without clustermesh or not, daemon will update node local ip addresses used for rama overlay network to `node.Annotations[constants.AnnotationNodeLocalVxlanIPList]`, which should also be updated to spec.endpointIPList of RemoteVtep CRs.

Reply:
yeah, i gotcha